### ISSUE TITLE
Segments colours are messed up using custom themes

### ISSUE THREAD

**Issue by jwintz** [domtronn/spaceline-all-the-icons.el#7]
Hello,

Great package ! Was waiting for it to get rid of my custom spaceline config ;-)

I've been giving a shot at your package, together with [doom-theme-one](https://github.com/hlissner/emacs-doom-theme).

They do not fit well, in that sense that segments are not coloured. Is there a specific face to set up ?

Also, I am quite surprised that projectile, and file name come as a single segment.

Finally, using doom-theme, the wave segment is not rendered correctly. I do not see anzu working whatsoever, and the clock icon is wrong. Here's the setup:

```emacs-lisp
(use-package spaceline-all-the-icons
  :after spaceline
  :config
  (spaceline-all-the-icons-theme)
  (spaceline-all-the-icons--setup-anzu)
  (spaceline-all-the-icons--setup-package-updates)
  (spaceline-all-the-icons--setup-paradox)
  (spaceline-all-the-icons--setup-neotree)
  (setq spaceline-all-the-icons-flycheck-alternate t)
  (setq spaceline-all-the-icons-highlight-file-name t)
  (setq spaceline-all-the-icons-separator-type (quote wave)))
```

<img width="1676" alt="screen shot 2017-04-16 at 19 56 01" src="https://cloud.githubusercontent.com/assets/3277054/25073109/6b067dbc-22df-11e7-85fe-27efb0ee8b40.png">

Cheers,

  - ju

**Comment by domtronn** [comment]
Hi @jwintz, thanks! Good news, I can reproduce this myself 😅 

_w.r.t._ the segment colours, yes, they use specific faces for each segment, these include

* The result of `(funcall spaceline-highlight-face-func)` _most likely `spaceline-highlight-face`_
* `powerline-active1`
* `powelrine-active2`
* `mode-line`
* `powerline-inactive1`
* `powerline-inactive2`

And since the wave separator is basically an icon, it calculates it's _foreground_ and _background_ colours from the two faces its transitioning between, _e.g._ from `spaceline-highlight-face` to `powerline-active-1`... The problem here, is that `spaceline-highlight-face` doesn't have a `background` property set on it, so when you call

```el
(face-background 'spaceline-highlight-face)
```

It returns `nil` rather than the _inherited_ background colour and so then defaults to inheriting the foreground colour, which is wrong!

_w.r.t._ the projectile/file-name segments, they're still separate segments 😃 they're just very coupled to try and produce a "smart" behaviour, that is,

* When inside a project, file path will show _relative_ to the _project root_
* When `projectile` is disabled  or outside a project or no project is set, file path will show _relative to `(getenv "HOME")`

And these are controlled by `all-the-icons-projectile`, `all-the-icons-buffer-path` and `all-the-icons-buffer-id` - You should be able to toggle these on and off independently to suit your taste and if you have any suggestions for the behaviour, I'm more than happy to hear them 😃 

Lastly, I'm not sure why the Clock Icon isn't showing? 
Do you have [`weather-icons`](https://github.com/domtronn/all-the-icons.el/blob/master/fonts/weathericons.ttf) installed? 
What happens if you call `all-the-icons-insert-icons-for-wicon` inside a `*scratch*` buffer?

**tl;dr**
Yep, from what I can tell from your set up, you will need to set `spaceline-highlight-face` & `powerline-active2` to have background colours.

Try toggling `spaceline-toggle-all-the-icons-projetcile`, `spaceline-toggle-all-the-icons-buffer-path` and `spaceline-toggle-all-the-icons-buffer-id` on and off to see how the file path works

Please try calling `all-the-icons-insert-icons-for-wicon` inside a `*scratch*` buffer and screenshot the results?

**Comment by domtronn** [comment]
Also, I forgot to mention about `anzu-mode`...  Do you have `global-anzu-mode` enabled? And what's the value of `anzu-mode-line-update-function` after your init? It could be that `anzu` is loading _after_ `spaceline-all-the-icons` and therefore overwriting back to default...?

**Comment by jwintz** [comment]
Hi,

Sorry for the noise, but I just remembered there use to be a segment for the `text-increase` count in the default spaceline setup. Is there any way to get it back ?

Many thanks !

**Comment by jwintz** [comment]
Will dig into segment faces soon. There was another point I wanted to bring up: with `spaceline-helm-mode` and `spaceline-info-mode`, using `spaceline-all-the-icons`, the modelling lowers in height as shown in the screenshot.

Is there a specific scale factor to set up ?

Cheers,

![screen shot 2017-04-20 at 16 56 18](https://cloud.githubusercontent.com/assets/3277054/25237397/d0857766-25ea-11e7-961e-9266581d1124.png)

**Comment by jwintz** [comment]
Hi there,

Some answers to my own questions. 

Here's my current config:

```emacs-lisp
(use-package spaceline-all-the-icons
  :after spaceline
  :config
  (setq spaceline-all-the-icons-icon-set-flycheck-slim (quote dots))
  (setq spaceline-all-the-icons-icon-set-git-ahead (quote commit))
  ;; (setq spaceline-all-the-icons-icon-set-window-numbering (quote square))
  (setq spaceline-all-the-icons-flycheck-alternate t)
  (setq spaceline-all-the-icons-highlight-file-name t)
  (setq spaceline-all-the-icons-separator-type (quote none))
  (spaceline-all-the-icons-theme)
  (spaceline-all-the-icons--setup-anzu)
  (spaceline-all-the-icons--setup-package-updates)
  (spaceline-all-the-icons--setup-paradox)
  (spaceline-all-the-icons--setup-neotree)
  (spaceline-toggle-all-the-icons-vc-icon-on)
  (spaceline-toggle-all-the-icons-fullscreen-on)
  (spaceline-toggle-all-the-icons-flycheck-status-on)
  (spaceline-toggle-all-the-icons-git-status-on)
  (spaceline-toggle-all-the-icons-vc-icon-on)
  (spaceline-toggle-all-the-icons-mode-icon-on)
  (spaceline-toggle-all-the-icons-package-updates-on)
  (spaceline-toggle-all-the-icons-text-scale-on)
  (spaceline-toggle-all-the-icons-region-info-on))
```

* package icons shows up just fine *after* `M-x package-refresh-contents`, which makes sense, otherwise it would slow down the startup process
* text-scale works fine whenever you increase or decrease the text scale. By default the modeline does not render anything with the following message: `Error during redisplay: (eval (spaceline-ml-all-the-icons)) signaled (void-variable text-scale-mode-lighter) [36 times]`
* using `(setq spaceline-all-the-icons-icon-set-window-numbering (quote square))`, breaks the modeline rendering when window number exceeds one: `Invalid face attribute :background nil
Invalid face attribute :foreground nil
Error during redisplay: (eval (spaceline-ml-all-the-icons)) signaled (wrong-type-argument listp (:height 1.2 :inherit)) [24 times]`
* `(spaceline-toggle-all-the-icons-vc-icon-on)` has no effect, would love to see on what branch I am

To be continued, love `spaceline-all-the-icons` !

**Comment by domtronn** [comment]
Hey @jwintz, thanks for all the useful information! I genuinely don't mind you asking lots of questions so please keep going.

I hope you don't mind, but I was going to split this into separate issues - just to help me compartmentalise the information and focus on an answer to each one individually. This should hopefully help other users who have similar problems as it should remove noise.

I'll quote the segments of this issue and link them back to this 👍

**Comment by jwintz** [comment]
Thanks for being so kind. I am slowly getting familiar with to code, so I'll hope to contribute soon.

**Comment by FieryCod** [comment]
@jwintz Is there any chance that you will show us your `.spacemacs` file? You configuration is pretty amazing. Could you gist your config please?

**Comment by jwintz** [comment]
Sure. I'm not using spacemacs. Here my config: https://github.com/jwintz/prolusion/tree/develop.
